pyramid/csrf.py

```diff
@@ -0,0 +1,225 @@
+    """
+    registry = request.registry
+    csrf = registry.queryUtility(ICSRF)
+    if csrf is not None:
+        return csrf.get_csrf_token(request)
+
+
+def new_csrf_token(request):
+    """ Generate a new CSRF token for the request passed and persist it in an
+    implementation defined manner. This calls the equivalent method in the
+    chosen CSRF protection implementation.
+
+    .. versionadded :: 1.8a1
+    """
+    registry = request.registry
+    csrf = registry.queryUtility(ICSRF)
+    if csrf is not None:
+        return csrf.new_csrf_token(request)
+
+
+def check_csrf_token(request,
+                     token='csrf_token',
+                     header='X-CSRF-Token',
+                     raises=True):
+    """ Check the CSRF token in the request's session against the value in
+    ``request.POST.get(token)`` (if a POST request) or
+    ``request.headers.get(header)``. If a ``token`` keyword is not supplied to
+    this function, the string ``csrf_token`` will be used to look up the token
+    in ``request.POST``. If a ``header`` keyword is not supplied to this
+    function, the string ``X-CSRF-Token`` will be used to look up the token in
+    ``request.headers``.
+
+    If the value supplied by post or by header doesn't match the value
+    supplied by ``request.session.get_csrf_token()``, and ``raises`` is
+    ``True``, this function will raise an
+    :exc:`pyramid.exceptions.BadCSRFToken` exception.
+    If the values differ and ``raises`` is ``False``, this function will
+    return ``False``.  If the CSRF check is successful, this function will
+    return ``True`` unconditionally.
+
+    Note that using this function requires that a :term:`session factory` is
+    configured.
+
+    See :ref:`auto_csrf_checking` for information about how to secure your
+    application automatically against CSRF attacks.
+
+    .. versionadded:: 1.4a2
+
+    .. versionchanged:: 1.7a1
+       A CSRF token passed in the query string of the request is no longer
+       considered valid. It must be passed in either the request body or
+       a header.
+
+    .. versionchanged:: 1.8a1
+       Moved from pyramid.session to pyramid.csrf
+    """
+    supplied_token = ""
+    # If this is a POST/PUT/etc request, then we'll check the body to see if it
+    # has a token. We explicitly use request.POST here because CSRF tokens
+    # should never appear in an URL as doing so is a security issue. We also
+    # explicitly check for request.POST here as we do not support sending form
+    # encoded data over anything but a request.POST.
+    if token is not None:
+        supplied_token = request.POST.get(token, "")
+
+    # If we were unable to locate a CSRF token in a request body, then we'll
+    # check to see if there are any headers that have a value for us.
+    if supplied_token == "" and header is not None:
+        supplied_token = request.headers.get(header, "")
+
+    impl = request.registry.getUtility(ICSRF)
```

Comment:
As this one is `getUtility` it will raise a `ComponentLookupError` if it's missing, were you thinking of something a bit more descriptive?